src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        final Node<T> prevNode = last;
+        final Node<T> newNode = new Node<>(prevNode, value, null);
+        last = newNode;
+        if (prevNode == null) {
+            first = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Can't element index " + index);
+        }
```

Comment:
```suggestion
```
We do not need it. if (index == size) - everything is ok. If not - we call `checkElementIndex(int index)` inside `getNode(index);` on line 32.